feat: add monorepo

tobySolutions opened the pull request:
<!-- Please follow this pattern for generating pull requests -->

## Description of the pull request

This is the pull request to integrate Mono repo support for the bun project. I followed two resources to achieve this current version in my pull request:

- https://bun.sh/guides/install/workspaces
- https://github.com/colinhacks/bun-workspaces

<!-- Provide a brief description of the changes and what it is addressing. -->
<!-- Please do specify the issue number clearly -->

## Changes made

<!-- Provide brief information about the changes made -->

- Setup the root package.json
- Created the packages folder and move the backend code to a folder called `sprout-server`
- add the `.gitignore` file to the root
- ....

## Related issues

<!-- Provide the related issues that are being solved or addressed by this PR (if any) -->

## Testings done

<!-- Provide a happy path to test and reproduce this issue-->

## Screenshots (if any)

<!-- Provide the screenshots of the changes made and how it affects the project -->

## Checklist

- [ ] I have written tests
- [ ] My code does not produce new errors
- [ ] I gave myself a code review before asking others.


Please help review @GuiBibeau, @chin-flags

cxalem commented:
I wonder if we can use [turborepo](https://turbo.build/) with Bun 🤔

tobySolutions commented:
> I wonder if there's a way to use [turborepo](https://turbo.build/) with Bun 🤔

I think we can actually, but, would we be using turborepo to handle the mono-repo support?

cxalem commented:
> > I wonder if there's a way to use [turborepo](https://turbo.build/) with Bun 🤔
> 
> I think we can actually, but, would we be using turborepo to handle the mono-repo support?

This is really good, it'll work for now! Turborepo is an option to scale our monorepo, we could implement it later without a problem if we decide it's needed. I would like to know @GuiBibeau and @chin-flags opinions also! 😅

tobySolutions commented:
> > > I wonder if there's a way to use [turborepo](https://turbo.build/) with Bun 🤔
> > 
> > 
> > I think we can actually, but, would we be using turborepo to handle the mono-repo support?
> 
> This is working really good already! Turborepo is an option to scale our monorepo, but we could implement it later without a problem if we decide it's needed. I would like to know @GuiBibeau and @chin-flags opinions also! 😅

Yeah, I see your point actually. I'd love to hear their opinions as well.

GuiBibeau commented on package.json:
```diff
@@ -1,26 +1,7 @@
+    "private": true,
+    "workspaces": [
+      "packages/*"
+    ]
```
Let's write some scripts that will call the API scripts so that we don't have to CD down deeper to run the project:

```
"scripts": {
  "api:dev": "cd packages/sprout-server && bun run dev"
}

```

maybe this thread can give insights? https://github.com/oven-sh/bun/issues/2232

tobySolutions replied:
Thanks for the thread by the way! I learned a lot!! I have done this.

GuiBibeau commented on packages/sprout-server/package.json:
```diff
@@ -0,0 +1,26 @@
+  "version": "1.0.50",
+  "module": "src/indes.ts",
+  "scripts": {
+    "dev:db:up": "docker-compose up -d",
+    "dev:db:down": "docker-compose down",
```
both of these are not in use anymore, let's remove them and the docker files.

tobySolutions replied:
I have removed these. Yeah, I don't think we need them anymore actually.

GuiBibeau reviewed (changes requested):
This if 🔥 🎸 

two  small changes requested:

- Let's `rm -rf .DS_Store` to ensure we don't track the file.
- removing the docker commands and files

tobySolutions commented:
> This if 🔥 🎸
> 
> two small changes requested:
> 
> * Let's `rm -rf .DS_Store` to ensure we don't track the file.
> * removing the docker commands and files

These have been resolved. I removed the file, added the scripts and removed the docker configs.
